routes/usersActivity.router.js

```diff
@@ -0,0 +1,234 @@
+      res.json({ success: true, data });
+    } catch (err) {
+      res
+        .status(500)
+        .json({
+          success: false,
+          message: "Unable to find user activity",
+          errorMessage: err.message,
+        });
+    }
+  })
+
+  .post(async (req, res) => {
+    try {
+      const usersActivity = req.body;
+      const NewUserActivity = new UsersActivity(usersActivity);
+      const savedUserActivity = await NewUserActivity.save();
+      res.json({ success: true, UsersActivity: savedUserActivity });
+    } catch (err) {
+      res
+        .status(500)
+        .json({
+          success: false,
+          message: "unable to register",
+          errorMessage: err.message,
+        });
+    }
+  });
+
+router.route("/wish").post(async (req, res) => {
+  try {
+    const usersActivity = req.body;
+    const NewUserActivity = new UsersActivity(usersActivity);
+    const savedUserActivity = await NewUserActivity.save();
+    res.json({ success: true, UsersActivity: savedUserActivity });
+  } catch (err) {
+    res
+      .status(500)
+      .json({
+        success: false,
+        message: "unable to register",
+        errorMessage: err.message,
+      });
+  }
+});
+
+router.route("/user/:id").get(async (req, res) => {
+  try {
+    const id = req.params.id;
+    const data = await UsersActivity.findById(id);
+    res.json({ success: true, data });
+  } catch (err) {
+    res.status(500).json({
+      success: false,
+      message: "Unable to populate user activity",
+      errorMessage: err.message,
+    });
+  }
+});
+
+/* check if wishlist id is present or not */
+router.route("/user/:id/wish/:wid").get(async (req, res) => {
+  try {
+    const { id, wid } = req.params;
+    const data = await UsersActivity.findOne({ _id: id });
+    const wishlistData = _.some(
+      data.wishlist,
+      (wish) => wish.toString() === wid
+    );
+    if (wishlistData) {
+      res.json({ success: true });
+    } else {
+      res.json({ success: false });
+    }
+  } catch (err) {
+    res.status(500).json({
+      success: false,
+      message: "Unable to find wish activity",
+      errorMessage: err.message,
+    });
+  }
+});
+
+//update wish
+router.route("/user/:id/wish/update/:wid").post(async (req, res) => {
+  try {
+    const id = req.params.id;
+    const wid = [req.params.wid];
+    const data = await UsersActivity.findOne({ _id: id });
+    const wishlistData = _.extend(data, {
+      wishlist: _.union(data.wishlist, wid),
+    });
+    await wishlistData.save();
+    res.json({ success: true, status:200 });
+  } catch (err) {
+    res.status(500).json({
+      success: false,
+      message: "Unable to find wish activity",
+      errorMessage: err.message,
+    });
+  }
+});
+
+/* check if wishlist id is present remove it */
+router.route("/user/:id/wish/:wid").delete(async (req, res) => {
+  try {
+    const { id, wid } = req.params;
+    const data = await UsersActivity.findById(id);
```

Comment:
`wishlistData` can be deleted without filtering on wishlist. mongoose has in-built function for it.